### PR TITLE
Warn if initrd cannot be created when udev not there

### PR DESCRIPTION
**Pull request by jsmeix**
Warn if initrd cannot be created when udev not there
and some more code cleanup, see
https://github.com/rear/rear/pull/1171#issuecomment-274442700

### CODE REVIEW

**Comment by jsmeix** [comment]
@gdha 
currently this one is not at all tested by me.
I will test it tomorrow.
I only did that pull request soon so that
you could have a first look if you like.

File: usr/share/rear/finalize/Debian/i386/170_rebuild_initramfs.sh [hunk 1]
```diff
@@ -1,71 +1,87 @@
...
-	# -t " " -k 1 tries to keep the comments unsorted
+# Skip if there is nothing to do.
+# During "rear recover" 260_recovery_storage_drivers.sh creates $TMP_DIR/storage_drivers
+if ! test -s $TMP_DIR/storage_drivers ; then
+    Log "Skip recreating initramfs: No needed storage drivers ('$TMP_DIR/storage_drivers' is empty)"
+    return 0
+fi
+# During "rear mkbackup/mkrescue" 260_storage_drivers.sh creates $VAR_DIR/recovery/storage_drivers
+if cmp -s $TMP_DIR/storage_drivers $VAR_DIR/recovery/storage_drivers ; then
+    Log "Skip recreating initramfs: '$TMP_DIR/storage_drivers' and '$VAR_DIR/recovery/storage_drivers' are same"
```
Comment (gdha) [inline usr/share/rear/finalize/Debian/i386/170_rebuild_initramfs.sh]: perhaps `are same` should be `are the same`?

**Comment by gdha** [comment]
@jsmeix Wait for the pull request 'till you have tested it yourself first.

File: usr/share/rear/finalize/Debian/i386/170_rebuild_initramfs.sh (hunk 1 above)
Comment (jsmeix) [inline usr/share/rear/finalize/Debian/i386/170_rebuild_initramfs.sh]: I even asked a native English speaking colleague
and you are right:
It is like "foo and bar are the same"
versus "foo and bar are same as baz".

**Comment by jsmeix** [comment]
FYI regarding my code cleanup:
I documented the basic idea behind in the new section
"Return early, return often" in
https://github.com/rear/rear/wiki/Coding-Style

**Comment by jsmeix** [comment]
Does not yet work because
'cmp' is missing in the recovery system:
<pre>
+ source /usr/share/rear/finalize/SUSE_LINUX/i386/170_rebuild_initramfs.sh
++ test -s /tmp/rear.ZrH4yz5LjAGGsMC/tmp/storage_drivers
++ cmp -s /tmp/rear.ZrH4yz5LjAGGsMC/tmp/storage_drivers /var/lib/rear/recovery/storage_drivers
/usr/share/rear/finalize/SUSE_LINUX/i386/170_rebuild_initramfs.sh: line 11: cmp: command not found
</pre>

**Comment by jsmeix** [comment]
Damn!
Now I have messed up this pull request.

**Comment by jsmeix** [comment]
I did a new (and hopefully clean) pull request:
https://github.com/rear/rear/pull/1179

**Pull request closed without merging**